### PR TITLE
Dockerfile: change to the alpine image

### PR DESCRIPTION
**Pull request by eterny13**
https://github.com/sugarlabs/musicblocks-v4-lib/issues/87  doesn't have "needs help", but I've fixed it.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,25 +1,16 @@
-FROM debian:buster-slim
+FROM alpine:latest
```
Comment (meganindya) [inline Dockerfile:1]: It's better to switch to the alpine based Node.js image itself.
Please update to `FROM node:lts-alpine` (and specify the version of the node package on top as a comment).

File: Dockerfile [hunk 2]
```diff
@@ -1,25 +1,16 @@
...
-# install node.js:14.x
-RUN apt update --no-install-recommends -yq \
-    && apt-get install curl gnupg -yq \
-    && curl -sL https://deb.nodesource.com/setup_14.x | bash \
-    && apt-get install nodejs -yq \
-    && apt-get clean -y \
-    && rm -rf /var/lib/apt/lists/*
+RUN apk update \
+    && apk add curl gnupg \
+    && apk add nodejs npm
```
Comment (meganindya) [inline Dockerfile:5]: Wouldn't require this if we use the `node` image itself.

File: Dockerfile [hunk 3]
```diff
@@ -1,25 +1,16 @@
...
 
-# update npm
-RUN npm install -g npm
```
Comment (meganindya) [inline Dockerfile:12]: Please retain this. This one updates `npm` to the latest version.

File: Dockerfile [hunk 4]
```diff
@@ -1,25 +1,16 @@
...
-
-# install typescript compiler
```
Comment (meganindya) [inline Dockerfile:14]: Please retain comment.

File: Dockerfile [hunk 5]
```diff
@@ -1,25 +1,16 @@
...
 RUN npm install -g typescript
 
-# install ts-node (to run/debug .ts files without manual transpiling)
```
Comment (meganindya) [inline Dockerfile:17]: Please retain comment.

File: Dockerfile [hunk 6]
```diff
@@ -1,25 +1,16 @@
...
 RUN npm install -g ts-node
 
-LABEL org.opencontainers.image.description='An initial development image based on a slimmed Debian \
-10.7 (buster), and further configured with Node.js v14, TypeScript compiler, and ts-node. This \
-does not contain any source files.'
+LABEL org.opencontainers.image.description='An initial development image based on the official \
+    Node.js (on Alpine LTS) image, and further configured with a HTTP server, TypeScript compiler, \
+    and ts-node. This is merely to provide an execution sandbox and does not contain source files.'
```
Comment (meganindya) [inline Dockerfile:13]: Please update to `LABEL org.opencontainers.image.description='An initial development image based on the official Node.js (on Alpine LTS) image, and further configured with TypeScript compiler and ts-node. This is merely to provide an execution sandbox  (for bind mounting source files) and does not contain source files.'`

File: Dockerfile [hunk 7]
```diff
@@ -1,25 +1,16 @@
...
 
 WORKDIR /app
```
Comment (meganindya) [inline Dockerfile:16]: Remove port expose.
Instead add `CMD [ "sh" ]`.

**Review by meganindya — CHANGES_REQUESTED** [review]

**Comment by eterny13** [comment]
@meganindya 
Thanks!, I've reflected comments.

**Pull request merged**